client/lxd_instances.go

```diff
@@ -1125,7 +1125,14 @@ func (r *ProtocolLXD) ExecInstance(instanceName string, exec api.InstanceExecPos
 	}
 
 	// Send the request
-	op, _, err := r.queryOperation("POST", uri, exec, "", false)
+	var op Operation
+	var err error
+	if r.CheckExtension("client_operation_wait") == nil {
```

Comment:
Could you use 

```go
useEventListener := r.CheckExtension("client_operation_wait") != nil
op, _, err = r.queryOperation("POST", uri, exec, "", useEventListener)
```
?

Reply:
Also, when did the `/wait` endpoint get introduced in the main API, does it already have a API extension?

Reply:
Finally, does this work for LXD talking to the lxd-agent? Does it returns a list of API extensions the server had when it was started?

Reply:
There is no API extension for operation wait as far as I can tell.The LXD agent returns the same list of API extensions as LXD does so this works for both the agent and the server.

Reply:
That being said, it is quite misleading for the lxd-agent to return the full list of API extensions that the server has, since it clearly doesn't implement the full API...

Reply:
Mmm yeah tho at least for this use case it works. To be clear the agent returns the list it had start time and doesn't proxy request back to lxd?

Reply:
It will return the static contents of `version.APIExtensions` at the time it was compiled. So older versions of the agent won't have the new list.

Reply:
perfect